test/common/scale_up_replicas_threescale.go

```diff
@@ -0,0 +1,166 @@
+package common
+
+import (
+	goctx "context"
+	"fmt"
+	"testing"
+	"time"
+
+	threescalev1 "github.com/3scale/3scale-operator/pkg/apis/apps/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	numberOfReplicas  int64 = 2
+	scaleUpReplicas   int64 = 3
+	scaleDownReplicas int64 = 1
+	name                    = "3scale"
+	namespace               = "redhat-rhmi-3scale"
+	retryInterval           = time.Second * 20
+	timeout                 = time.Minute * 7
+)
+
+func TestReplicasInThreescale(t *testing.T, ctx *TestingContext) {
+
+	apim, err := getAPIManager(ctx.Client)
+	if err != nil {
+		t.Fatalf("failed to get APIManager : %v", err)
+	}
+
+	t.Log("Checking correct number of replicas are set")
+	if err := checkNumberOfReplicasAgainstValue(apim, ctx, numberOfReplicas, retryInterval, timeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas to start : %v", err)
+	}
+
+	apim, err = updateAPIManager(ctx.Client, scaleUpReplicas)
+	if err != nil {
+		t.Fatalf("Unable to update : %v", err)
+	}
+
+	t.Log("Checking correct number of updated replicas are set")
+	if err := checkNumberOfReplicasAgainstValue(apim, ctx, scaleUpReplicas, retryInterval, timeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas : %v", err)
+	}
+
+	apim, err = updateAPIManager(ctx.Client, scaleDownReplicas)
+	if err != nil {
+		t.Fatalf("Unable to update : %v", err)
+	}
+
+	t.Log("Checking correct number of updated replicas are set")
+	if err := checkNumberOfReplicasAgainstValue(apim, ctx, scaleDownReplicas, retryInterval, timeout, t); err != nil {
+		t.Fatalf("Incorrect number of replicas : %v", err)
+	}
+
+	apim, err = updateAPIManager(ctx.Client, numberOfReplicas)
+	if err != nil {
+		t.Fatalf("Unable to update : %v", err)
+	}
```

Comment:
I don't think we need to check the scale down replias were set. It will be enough to check they are reset to the minimum below